service/build.gradle

```diff
@@ -100,7 +100,7 @@ dependencies {
     testFixturesImplementation "org.springframework:spring-webmvc:${gradle.vSpringWeb}"
 
     // [Application DB] MySQL/MariaDB
-    implementation 'org.mariadb.jdbc:mariadb-java-client:3.4.0'
```

Comment:
There is an open bug in java clients 3.x (see https://mariadb.com/kb/en/unsupported-command-in-prepared-statement-using-the-new-jdbc-driver-303/ and https://jira.mariadb.org/browse/MDEV-26673). Batch updates are not working in 3.x versions. 

we have 2 options - 
- Downgrade the client to latest 2.x (verified on separate branch that this error is resolved). Lock file changes do not show major changes - only the client gets downgraded and most other lib changes are for the better - extra deps are removed. 
- Rewrite our code to only have single row updates. This change is not only for this PR (clone cohort) but others which may be using batch update. Going with single row updates is potentially a blocker for performance reasons, and we don't ideally want to that (bad design/implementation). 

hence not choosing option 2. We can upgrade it back again when the client issue is resolved

Reply:
Security - accorinding to maven repo, this version has a security vuln due to version of included logback library (see https://mvnrepository.com/artifact/org.mariadb.jdbc/mariadb-java-client/2.7.12). however we have a newer version of logback already, so i would not expect a new security issue.

Reply:
Compatibility: would not expect any. We had a 2.7.* version until the recent java upgrade https://github.com/DataBiosphere/tanagra/blob/6f8f86e40e17f796aa3bac200a38d13ea399f029/service/gradle.lockfile#L269